### PR TITLE
Move LR protobufs shared with runtime from infrastructure to runtime.

### PR DESCRIPTION
**Pull request by pankti-m**
## Overview
Some protobufs and constants are required in the runtime package.  Adding a dependency from runtime -> infrastructure will result in a circular dependency.  So move the required protobufs and constants in the runtime package.

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/proto/log_replication_metadata.proto [hunk 1]
```diff
@@ -57,60 +57,18 @@ message ReplicationMetadata {
...
-  SyncType syncType = 1;
-  SyncStatus status = 2;
-  SnapshotSyncInfo snapshotSyncInfo = 3;
+enum SyncStatus {
+  ONGOING = 0;
+  COMPLETED = 1;
+  ERROR = 2;
+  NOT_STARTED = 3;
+  STOPPED = 4;
+  UNAVAILABLE = 5;
```
Comment (xcchang) [inline infrastructure/proto/log_replication_metadata.proto:71]: nit: I thought we can leave a comment here explaining something like V1 schemas are also moved to runtime/, keeping the old schema here for some rolling upgrade consideration?

**Review by xcchang — APPROVED** [review]
LGTM, just 1 minor comment

File: infrastructure/proto/log_replication_metadata.proto (hunk 1 above)
Comment (xcchang) [reply infrastructure/proto/log_replication_metadata.proto:71]: Sorry I mixed it up a little bit, I was thinking about the change we did for master before where we moved the V1 schema to runtime/ as well ... so please ignore this and I'm find with the change :)

**Review by hisundar — APPROVED** [review]
LGTM, let's work on adding the data-migration task for tables like the ReplicationStatusTable so all its contents can be re-written, serialized on runtime based protobufs

**Review by annym — APPROVED** [review]

**Pull request merged**